File: ios/Sources/Domain/UseCases/ReadChapterUseCase.swift
import Foundation

struct ReadChapterUseCase: Sendable {
    let repository: BibleRepository

    func callAsFunction(version: BibleVersion, book: BibleBook, chapter: Int) async throws -> [BibleVerse] {
        try await repository.readChapter(version: version, book: book, chapter: chapter)
    }
}

File: android/app/src/main/java/com/soobinpark/appcraft/readingbible/domain/usecase/ReadChapterUseCase.kt
package com.soobinpark.appcraft.readingbible.domain.usecase

import com.soobinpark.appcraft.readingbible.domain.model.BibleBook
import com.soobinpark.appcraft.readingbible.domain.model.BibleVerse
import com.soobinpark.appcraft.readingbible.domain.model.BibleVersion
import com.soobinpark.appcraft.readingbible.domain.repository.BibleRepository

class ReadChapterUseCase(
    private val repository: BibleRepository,
) {
    suspend operator fun invoke(
        version: BibleVersion,
        book: BibleBook,
        chapter: Int,
    ): List<BibleVerse> = repository.readChapter(version, book, chapter)
}
